core/controllers/profile.py

```diff
@@ -519,14 +518,24 @@ class UsernameCheckHandler(base.BaseHandler):
 
     REDIRECT_UNFINISHED_SIGNUPS = False
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'username': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_valid_username_string'
+                    }]
+                }
+            }
+        }
+    }
+
     @acl_decorators.require_user_id_else_redirect_to_homepage
     def post(self):
         """Handles POST requests."""
-        username = self.payload.get('username')
-        try:
-            user_domain.UserSettings.require_valid_username(username)
-        except utils.ValidationError as e:
-            raise self.InvalidInputException(e)
+        username = self.normalized_payload.get('username')
```

Comment:
@vojtechjelinek In the profile_test file, the validation error of `username` is expected from function `require_valid_username`. In current codes, `username` is validated through the schema, which causes a backend test failure as follows. Do you think it is good to delete the validator `is_valid_username_string` in the schema and only validate `username` in `post` with `require_valid_username` as previous codes? 
<img width="1041" alt="Screen Shot 2022-06-27 at 2 16 50 PM" src="https://user-images.githubusercontent.com/31307813/176018649-1ce4b467-209a-489f-811e-e2a2c42b08b6.png">

Reply:
Just fix the test so that the assertion expects different message.

Reply:
Thanks